tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/NetStandardLibraries/LogInstrumentation/NLogLoggingAdapter.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using NLog;
+using NLog.Layouts;
+using NLog.Targets;
+
+namespace MultiFunctionApplicationHelpers.NetStandardLibraries.LogInstrumentation
+{
+    class NLogLoggingAdapter : ILoggingAdapter
+    {
+        private static Logger _log;
+
+        public NLogLoggingAdapter()
+        {
+        }
+
+        public void Debug(string message)
+        {
+            _log.Debug(message);
+        }
+
+        public void Info(string message)
+        {
+            _log.Info(message);
+        }
+
+        public void Warn(string message)
+        {
+            _log.Warn(message);
+        }
+
+        public void Error(string message)
+        {
+            _log.Error(message);
+        }
+
+        public void Fatal(string message)
+        {
+            _log.Fatal(message);
+        }
+
+        public void Configure()
+        {
+            var logconsole = new ConsoleTarget();
+        
+            NLog.Config.SimpleConfigurator.ConfigureForTargetLogging(logconsole, LogLevel.Debug);
```

Comment:
Maybe use isolated NLog.LogFactory like this:
```c#
var logConsole = new ConsoleTarget();
if (layoutOverride != null)
      logConsole = layoutOverride;
var logFactory = new NLog.LogFactory();
var logConfig  = new NLog.Config.LoggingConfiguration()
logConfig.AddTarget("console", logConsole);
logConfig.LoggingRules.Add(new NLog.Config.LoggingRule("*", NLog.LogLevel.Trace, logConsole));
logFactory.Configuration = logConfig;
return logFactory.GetLogger("LoggingTest");
```
Then `ConfigureJsonLayoutAppenderForDecoration()` can provide its JsonLayout as `layoutOverride`.

Reply:
Thank you for this!  I updated the adapter to setup the log this and all the tests passed!